sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs

```diff
@@ -26,7 +26,11 @@ public RecordMatcher(RecordedTestSanitizer sanitizer)
             "x-ms-date",
             "x-ms-client-request-id",
             "User-Agent",
-            "Request-Id"
+            "Request-Id",
+            "If-Match",
```

Comment:
I'm not sure about this, `If*` headers are something that clients generate based on inputs. What is the reason you would like to avoid matching them?

Reply:
I guess because of equivalency check?

Reply:
They're always ETags or Dates.  I can move them to just the StorageRecordMatcher if you don't think they should generally be ignored.

Reply:
Separated `ExcludedHeaders` for matching and `VolatileHeaders` for ignoring unchanged values on re-record.